### PR TITLE
Route user profile

### PR DESCRIPTION
**Pull request by NandiniGera**
When we click on the username who has made a comment, it routes to that person's profile now!
I've tried working it on many users as well
Completes issue #796

### CODE REVIEW

**Comment by NandiniGera** [comment]
Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment.
I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

**Comment by narayan954** [comment]
> Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.

**Comment by narayan954** [comment]
> Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

reason it's showing your profile data is because it's not getting external data and defaulting to logged in user data.
The best way to solve this issue imo is to implement dynamic routes asap for profile, so we can just redirect to the dynamic route of that user, and every other detail should be taken care of independently by that page

File: src/components/Post/CommentDialogBox.jsx [hunk 1]
```diff
@@ -44,15 +45,18 @@ const CommentDialogBox = ({
...
-                    <span className="comment-doer-name">
-                      {userComment.content.username}
-                    </span>
-                  </Link>
+                <div className="comment-doer">
+                <AccountCircleIcon />
+                <span className="comment-doer-name" onClick={() => {
+                  navigate("/dummygram/profile", {
+                    state: {
+                      name: userComment.content.username,
```
Comment (narayan954) [inline src/components/Post/CommentDialogBox.jsx:53]: photo and other details can be passed in this state object

**Review by narayan954 — CHANGES_REQUESTED** [review]
I've marked where the problem currently is

File: src/components/Post/CommentDialogBox.jsx (hunk 1 above)
Comment (NandiniGera) [reply src/components/Post/CommentDialogBox.jsx:53]: > photo and other details can be passed in this state object

I tried doing that, but it didn't work, thats why i removed that code in the updated commit :/

image

**Comment by NandiniGera** [comment]
> > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> 
> reason it's showing your profile data is because it's not getting external data and defaulting to logged in user data. The best way to solve this issue imo is to implement dynamic routes asap for profile, so we can just redirect to the dynamic route of that user, and every other detail should be taken care of independently by that page.
> 
> Till then, we can pass the state as in other components to the profile page while naviagting! :D

oohhh okayy got it:)

**Comment by NandiniGera** [comment]
> I've marked where the problem currently is



> > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> 
> perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.

I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/

**Comment by narayan954** [comment]
> > I've marked where the problem currently is
> 
> > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > 
> > 
> > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> 
> I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/

listen, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal .
So you can just redirect to the correct username, rest is handled on its own :D

File: src/components/Post/CommentDialogBox.jsx (hunk 1 above)
Comment (narayan954) [reply src/components/Post/CommentDialogBox.jsx:53]: ohhhh, perhaps some other issue, or component getting refreshed on its own. But dont worry now, route has been made dynamic and you dont need to pass these states at all now!! Amazing, isn't it? :)

**Comment by NandiniGera** [comment]
> > > I've marked where the problem currently is
> > 
> > 
> > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > 
> > > 
> > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > 
> > 
> > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> 
> listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D

okayy thats great!

**Comment by narayan954** [comment]
> > > > I've marked where the problem currently is
> > > 
> > > 
> > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > 
> > > > 
> > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > 
> > > 
> > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > 
> > 
> > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> 
> okayy thats great!

Hi @NandiniGera is there any blocker?

**Comment by NandiniGera** [comment]
> > > > > I've marked where the problem currently is
> > > > 
> > > > 
> > > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > > 
> > > > > 
> > > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > > 
> > > > 
> > > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > > 
> > > 
> > > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> > 
> > 
> > okayy thats great!
> 
> Hi @NandiniGera is there any blocker?

Actually I updated the code, but it isn't working :/
I've added the commit

**Comment by narayan954** [comment]
> > > > > > I've marked where the problem currently is
> > > > > 
> > > > > 
> > > > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > > > 
> > > > > > 
> > > > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > > > 
> > > > > 
> > > > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > > > 
> > > > 
> > > > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> > > 
> > > 
> > > okayy thats great!
> > 
> > 
> > Hi @NandiniGera is there any blocker?
> 
> Actually I updated the code, but it isn't working :/ I've added the commit

aight checking

**Review by narayan954 — APPROVED** [review]
looks good to me!

**Pull request merged**